src/models/assetGroup/useAssetGroupSighting.js

```diff
@@ -1,54 +1,29 @@
-import { useEffect, useState } from 'react';
 import axios from 'axios';
-import { get } from 'lodash-es';
-import { formatError } from '../../utils/formatters';
+import { useQuery } from 'react-query';
 
-export default function useAssetGroupSighting(assetGroupSightingId) {
-  const [error, setError] = useState(null);
-  const [loading, setLoading] = useState(true);
-  const [statusCode, setStatusCode] = useState(null);
-  const [data, setData] = useState(null);
-  const [refreshCount, setRefreshCount] = useState(0);
-
-  function refresh() {
-    setRefreshCount(refreshCount + 1);
-  }
-
-  useEffect(
-    () => {
-      const fetchAssetGroupSightingData = async () => {
-        try {
-          const response = await axios.request({
-            url: `${__houston_url__}/api/v1/asset_groups/sighting/as_sighting/${assetGroupSightingId}`,
-            method: 'get',
-          });
-
-          const responseStatusCode = get(response, ['status']);
-          setStatusCode(responseStatusCode);
-          const successful = responseStatusCode === 200;
-          if (!successful) setError(formatError(response));
+import { getAGSQueryKey } from '../../constants/queryKeys';
 
-          setLoading(false);
-          setData(get(response, ['data']));
-        } catch (fetchError) {
-          const responseStatusCode = get(fetchError, [
-            'response',
-            'status',
-          ]);
-          setStatusCode(responseStatusCode);
-          console.error(
-            `Error fetching asset group sighting ${assetGroupSightingId}`,
-          );
-          console.error(fetchError);
-          setError(formatError(fetchError));
-          setLoading(false);
-        }
-      };
-
-      if (assetGroupSightingId) fetchAssetGroupSightingData();
+export default function useAssetGroupSighting(assetGroupSightingId) {
+  const result = useQuery(
+    getAGSQueryKey(assetGroupSightingId),
+    async () => {
+      return axios.request({
+        url: `${__houston_url__}/api/v1/asset_groups/sighting/as_sighting/${assetGroupSightingId}`,
+        method: 'get',
+      });
+    },
+    {
+      refetchInterval: 5000,
```

Comment:
Because the sighting moves through phases based on WBIA, (detection => curation => matching). So it's very possible for the backend phase to switch while the user is on the page and we want to show them that update without refreshing.